src/Microsoft.AspNet.Server.Kestrel/Http/Listener.cs

```diff
@@ -13,8 +13,10 @@ namespace Microsoft.AspNet.Server.Kestrel.Http
     /// </summary>
     public abstract class Listener : ListenerContext, IDisposable
     {
-        protected Listener(ServiceContext serviceContext) : base(serviceContext)
+        protected Listener(ServiceContext serviceContext) 
+            : base(serviceContext)
         {
+            Memory2 = new MemoryPool2();
```

Comment:
I want @halter73 's PR to rebase before causing larger code churn like that. That's why the original memory pool and this one are temporarily distinct from each other.